packages/eas-cli/src/metadata/utils/array.ts

```diff
@@ -0,0 +1,6 @@
+/**
+ * Filter the array using a truthy predicate, returning a type-safe truthy array.
+ */
+export function coalesce<T>(list: (T | undefined | null)[]): T[] {
+  return list.filter(Boolean) as T[];
+}
```

Comment:
- The function name is misleading. Usually, "coalesce" is used for applying fallback values to potentially falsy values, sth like `const coalesce = (x?: number) => x ?? 2137;`. I'd be explicit here and change the name to sth like "filterOutFalsyValues" (this can be too explicit, leaving it up to you).
- Don't use `Boolean()`. `Boolean(0) === false` :) 
  <img width="136" alt="Screenshot 2022-10-11 at 11 05 46" src="https://user-images.githubusercontent.com/5256730/195047789-00c5201f-bbf1-48ce-9da6-8942055c53d6.png">
- You don't need `as T[]` part, see below.
- Maybe move the function to the `expodash` folder? I can see it can be used in other places too.

```suggestion
/**
 * Filter the array using a truthy predicate, returning a type-safe truthy array.
 */
export function CHANGETHENAME<T>(list: (T | undefined | null)[]): T[] {
  return list.filter<T>((i): i is T =>i !== undefined && i !== null);
}
```

Reply:
How about naming it just "filter"? It's the underlying action, while doing some typescript cleanups.

I'll move it to expodash

Reply:
`filter` is fine